HW06-solid-atm/src/main/java/ru/gavrilov/atm/model/BankCellImpl.java

```diff
@@ -0,0 +1,48 @@
+package ru.gavrilov.atm.model;
+
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Deque;
+
+/**
+ * @author gavrilov-sv
+ * created on 20.05.2020
+ */
+public class BankCellImpl implements Bankcell {
+
+    private Nominal nominal;
+    private Deque<Banknote> money = new ArrayDeque<>();
```

Comment:
Можно было не вводить учет экземпляров купюр.
Оставить примерно как у вас  было с `  Map<Nominal, Long>`

```suggestion
    private Nominal nominal;
    private int count;
```

С одной стороны, по сравнению с первоначальным вариантом, кажется что большой разницы нет, но с другой сравните такой код
```java
Map<Nominal, Long> initCassette
```

и такой 
```java
List<Cell> initCassette
```
В первом случае код читается как "Мапа номинала (хорошо, что ключ Nominal, а не Long) и непонятного Long. Так лезу в код глубже (или в комментарии) и пытаюсь понять, что это за Long...ага! количество купюр".
Во втором - "Здесь список ячеек" 

В одной строчке подобного кода проблемы нет, но когда такого кода много, то это уже может стать проблемой при изучении/модификации подобного кода.